.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,42 @@
+on:
+  workflow_call:
+    inputs:
+      name:
+        required: true
+        type: string
+      command:
+        required: true
+        type: string
+
+jobs:
+  benchmarks:
+    name: ${{ inputs.name }}
+    runs-on: gha-mm-scale-set-ubuntu-22.04-amd64-med
+    container:
+      image: cimg/node:22.13-browsers
+    steps:
+      - name: Setup environment
+        uses: metamask/github-tools/.github/actions/setup-environment@caching
+        with:
+          should-cache-restore: ${{ vars.USE_CACHING }}
+
+      - name: Download artifact prep-build-test-webpack
+        uses: actions/download-artifact@v4
+        with:
+          path: ./dist/
+          pattern: prep-build-test-webpack
+          merge-multiple: true
+
+      # Xvfb initializes a virtual "screen" where the browser opens
+      - name: Configure Xvfb
+        run: Xvfb -ac :99 -screen 0 1280x1024x16 &
+
+      - name: Run the benchmark
+        run: ${{ inputs.command }}
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ inputs.name }}
+          path: test-artifacts/chrome/benchmark/
+          retention-days: 5
```

Comment:
The number of `retention-days` was chosen very arbitrarily, and maybe someone has an opinion

Reply:
These particular benchmark artifacts are very very small.  It's just some JSON data that gets zipped.  On one run, it's 823 bytes and 196 bytes.  But we also may want to look back at these numbers much later.  GitHub probably isn't the best platform for that though, something like Sentry or Segment is probably much better.